repm/api.go

```diff
@@ -176,6 +213,73 @@ func (conn *connection) dispatchPullProgress(reqBytes []byte) ([]byte, error) {
 	return mustMarshal(res), nil
 }
 
+func (conn *connection) newTransaction() int {
+	conn.transactionMutex.Lock()
+	defer conn.transactionMutex.Unlock()
+	txID := conn.transactionCounter
+	conn.transactionCounter++
+	conn.transactions[txID] = conn.db.NewTransaction()
+	return txID
+}
+
+func (conn *connection) dispatchBeginTransaction(reqBytes []byte) ([]byte, error) {
+	var req beginTransactionRequest
+	err := json.Unmarshal(reqBytes, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	txID := conn.newTransaction()
+
+	res := beginTransactionResponse{
+		TransactionID: txID,
+	}
+	return mustMarshal(res), nil
+}
+
+func (conn *connection) dispatchCloseTransaction(reqBytes []byte) ([]byte, error) {
+	var req closeTransactionRequest
+	err := json.Unmarshal(reqBytes, &req)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := conn.findTransaction(req.TransactionID)
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Close()
```

Comment:
might be slightly safer to remove it from the map first then close it on the idea that the map holds only open transactions? (same applies to commit below)